Sources/Architecture/X86/HardwareBreakpointManager.cpp

```diff
@@ -48,17 +48,33 @@ ErrorCode HardwareBreakpointManager::add(Address const &address, Type type,
   return super::add(address, type, size, mode);
 }
 
+ErrorCode HardwareBreakpointManager::remove(Address const &address) {
+  auto loc = std::find(_locations.begin(), _locations.end(), address);
+  if (loc != _locations.end()) {
+    _locations[loc - _locations.begin()] = 0;
+  }
+
+  return super::remove(address);
```

Comment:
Yeah that makes sense. When I read this code I assumed that the location was picked at `add()` time which is obviously not the case.